indico/web/client/js/react/components/time/TimePicker.jsx

```diff
@@ -0,0 +1,213 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2025 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import ComboBox from 'indico/react/components/ComboBox';
+import {Translate} from 'indico/react/i18n';
+
+import {timeString} from './prop_types.js';
+import {Time, timeList} from './time_value.js';
+
+import './TimePicker.module.scss';
+
+const DEFAULT_STEP_SIZE = 15; // minutes
+const TIME_FORMAT_PLACEHOLDER = {
+  '12h': '--:-- AM/PM',
+  '24h': '--:--',
+};
+
+function TimeOptionLabel({value, label, duration}) {
+  return (
+    <time dateTime={value}>
+      <span>{label}</span>
+      {duration ? <span>{duration}</span> : null}
+    </time>
+  );
+}
+
+TimeOptionLabel.propTypes = {
+  value: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  duration: PropTypes.string.isRequired,
+};
+
+function getOptions(currentValue, startTime, step, min, max, timeFormat) {
+  return timeList({
+    markCurrent: currentValue,
+    startTime,
+    step,
+    min,
+    max,
+    timeFormat,
+  }).map(option => {
+    const {label, time, duration, disabled} = option;
+
+    return {
+      value: label,
+      disabled,
+      label: <TimeOptionLabel value={time} label={label} duration={duration} />,
+    };
+  });
+}
+
+function rankTimeOption(_, keywordText, option) {
+  const optionDisabled = option.hasAttribute('aria-disabled');
+  const optionText = option.textContent.replace(/\s+/g, '').toLowerCase();
+  const optionValue = option.firstElementChild.dateTime;
+  const keywordValue = Time.fromString(keywordText).toString();
+
+  keywordText = keywordText.toLowerCase().replace(/\s+/, '');
+
+  let optionRank = 0;
+
+  if (keywordText && !optionDisabled) {
+    if (optionValue === keywordValue) {
+      optionRank = Infinity;
+    } else if (optionText.startsWith(keywordText)) {
+      optionRank = 10;
+    }
+  }
+
+  return optionRank;
+}
+
+export default function TimePicker({
+  value,
```

Comment:
This seems to be completely ignored; when editing a registration the time picker always shows `--:--`, even though there's a valid time string in this prop.